### ISSUE TITLE
Default peer address is hard-coded in chaincode shim

### ISSUE THREAD

**Issue by bcbrock** [hyperledger-archives/fabric#1304]
From fabric/core/chaincode/shim at line 129:

```
func getPeerAddress() string {
    if peerAddress != "" {
        return peerAddress
    }

    if peerAddress = viper.GetString("peer.address"); peerAddress == "" {
        // Assume docker container, return well known docker host address
        peerAddress = "172.17.42.1:30303"
    }

    return peerAddress
}
```

The hard-coded default peer address is a very bad assumption. The Docker ".1" address is the Docker bridge itself, not any peer, and there is no guarantee that Docker is bridged at 172.17.42.1. The Start() call should fail explicitly if the peer address is unknown, as using this address will lead to a certain, more obscure failure.

**Comment by christo4ferris** [comment]
@bcbrock good catch, and yes, very bad assumption. There are a bunch of similar functions in peer/config.go and specifically, Cache.Configuration.getLocalAddress. Does this apply here?

**Comment by bcbrock** [comment]
@christo4ferris I'm not sure what you mean  re: peer/config.go, Cache.Configuration.getLocalAddress. I don't see any hard-coded addresses in there.

**Comment by mastersingh24** [comment]
@gennadylaventman  - can you add this to the list of issue around connection management?

**Comment by gennadylaventman** [comment]
@mastersingh24 - I will.

**Comment by bcbrock** [comment]
There is a similar hard-coded default in fabric/core/chaincode/chaincode_support.go

```
const (
    // DefaultChain is the name of the default chain.
    DefaultChain ChainName = "default"
    // DevModeUserRunsChaincode property allows user to run chaincode in development environment
    DevModeUserRunsChaincode       string = "dev"
    chaincodeStartupTimeoutDefault int    = 5000
    chaincodeInstallPathDefault    string = "/opt/gopath/bin/"
    peerAddressDefault             string = "0.0.0.0:30303"
)

```

**Comment by lehors** [comment]
@bcbrock This issue was addressed by PR #1520 and can now be closed.

**Comment by bcbrock** [comment]
Thanks Arnaud @lehors
